pkg/ddl/partition.go

```diff
@@ -3415,18 +3494,26 @@ func (w *reorgPartitionWorker) GetCtx() *backfillCtx {
 	return w.backfillCtx
 }
 
-func (w *worker) reorgPartitionDataAndIndex(t table.Table, reorgInfo *reorgInfo) error {
-	// First copy all table data to the new partitions
+func (w *worker) reorgPartitionDataAndIndex(t table.Table, reorgInfo *reorgInfo) (err error) {
+	// First copy all table data to the new AddingDefinitions partitions
 	// from each of the DroppingDefinitions partitions.
-	// Then create all indexes on the AddingDefinitions partitions
-	// for each new index, one partition at a time.
+	// Then create all indexes on the AddingDefinitions partitions,
+	// both new local and new global indexes
+	// And last update new global indexes from the non-touched partitions
+	// Note it is hard to update global indexes in-place due to:
```

Comment:
Does it mean even for `reorg partition` op, we will remove all key/value pairs not only relevant partitions?

Reply:
Yes, even for reorganize partition, we will recreate a full new global index.

The reason is that there is a DDL state transition where the previous schema version sees the old partitions and the current schema version sees the new partitions, and a single global index cannot represent a row that would match two different partitions (one for old and one for new).

Reply:
Could you provide a state change graph include job state and global index state? Maybe it's more easier to understand it.

Reply:
@Defined2014 please check the comments I added [here](https://github.com/pingcap/tidb/pull/53277/files#diff-a558fe65fa5ae53cf0710c8d700cbfd155d3ba42f4ed85494c01f9c1d24bbe6bR2811-R2875), are they what you were looking for?